samza-azure/src/main/java/org/apache/samza/system/eventhub/producer/AsyncSystemProducer.java

```diff
@@ -139,7 +139,6 @@ public synchronized void send(String source, OutgoingMessageEnvelope envelope) {
 
     // Auto update the metrics and possible throwable when futures are complete.
     sendResult.handle((aVoid, throwable) -> {
-        pendingFutures.remove(sendResult);
```

Comment:
This isn't necessary anymore?

Reply:
This was not necessary event before because we clear up the pendingFutures on flush. But long term we need to fix this properly where we remove the futures on callback and flush doesn't need to clear. This is to support brooklin pipelines that don't call flush.